### DISCUSSION TITLE
k3s.service not starting while configuring Setting up a High-availability K3s Kubernetes Cluster for Rancher

### DISCUSSION THREAD

**Discussion by gmishra-snet** [k3s-io/k3s#5878] [Q&A] [answered]
**document i follow for setup-**
- [https://rancher.com/docs/rancher/v2.5/en/installation/resources/k8s-tutorials/ha-with-external-db/](url)

 **error shown by systemctl status k3s.service-** 
ExecStart=/usr/local/bin/k3s server --datastore-endpoint=mysql://rancher:rancher@tcp(ip:3306)/rancher (code=exited, status=1/FAILURE)

**error shown by journalctl -xeu k3s.service-**
+ /usr/bin/systemctl is-enabled --quiet nm-cloud-setup.service
Failed to get unit file state for nm-cloud-setup.service: No such file or directory
 time="2022-07-17T07:43:14Z" level=info msg="Starting k3s v1.24.3-rc1+k3s1 (990ba0e8)"
time="2022-07-17T07:43:14Z" level=info msg="Configuring mysql database connection pooling: maxIdleConns=2, maxOpenConns=0, connMaxLifetime=0s"
time="2022-07-17T07:43:14Z" level=info msg="Configuring database table schema and indexes, this may take a moment..."
 time="2022-07-17T07:43:14Z" level=info msg="Database tables and indexes are up to date"
 time="2022-07-17T07:43:14Z" level=info msg="Kine available at unix://kine.sock"
:time="2022-07-17T07:43:14Z" level=fatal msg="starting kubernetes: preparing server: bootstrap data already found and encrypted with different token"
k3s.service: Main process exited, code=exited, status=1/FAILURE

**Comment by brandond** [comment, marked as answer]
It looks like you have not specified the `--token`, and the database is not empty. When joining a new server to the cluster, or when pointing a server at a database that was already used for an existing cluster, you must pass the same token that was used on the first server.
